services/gitdiff/gitdiff.go

```diff
@@ -1228,31 +1232,42 @@ func GetDiffRangeWithWhitespaceBehavior(gitRepo *git.Repository, beforeCommitID,
 	ctx, cancel := context.WithTimeout(git.DefaultContext, time.Duration(setting.Git.Timeout.Default)*time.Second)
 	defer cancel()
 
-	var cmd *exec.Cmd
+	argsLength := 6
+	if len(whitespaceBehavior) > 0 {
+		argsLength++
+	}
+	if len(skipTo) > 0 {
+		argsLength++
+	}
+
+	diffArgs := make([]string, 0, argsLength)
```

Comment:
Also it seems to save some memory reallocation time, but the logic is a little strange. I think we can make a default `diffArgs` array, and just use `append` to extend the size dynamically.

Reply:
We commonly do this sort of arg calculation in modules/git.